.github/workflows/run-examples.yml

```diff
@@ -72,3 +72,17 @@ jobs:
           echo "$pasted" > $GITHUB_STEP_SUMMARY
         shell: "bash -xe {0}"
 
+  UnitTests:
+    runs-on: ubuntu-latest
+    container:
+      # Requires repo to have action access in package settings
+      image: ghcr.io/uq-pac/basil-dev:latest
+      credentials:
+        username: ${{ github.actor }}
+        password: ${{ secrets.github_token }}
+    steps:
+      - uses: actions/checkout@v4
+
+        # every text except SystemTests:
+        # ./mill test.testOnly '*' -- -t '' | tr -d ':' | ansi2txt | grep -vE '^SystemTests' 
+      - run: ./mill test.testOnly DSAMemoryRegionSystemTestsBAP IntrusiveListTest IntrusiveListPublicInterfaceTest PointsToTest DSAMemoryRegionSystemTestsGTIRB MemoryRegionTestsNoRegion AnalysisSystemTestsBAP AnalysisSystemTestsGTIRB DataStructureAnalysisTest IrreducibleLoop ParamAnalysisTests LiveVarsAnalysisTests IRTest CILVisitorTest InterpreterTests InvariantTest ProcedureSummaryTests UnimplementedTests ExtraSpecTests BitVectorAnalysisTests TaintAnalysisTests IndirectCallTests MemoryRegionTestsDSA
```

Comment:
We definitely do not want to include all of these in the GitHub action and running all tests simultaneously like this breaks things at present, a bunch of tests (the various SystemTest variations) get in each other's way.

None of the following are expected to pass at present (some are not ever expected to pass) and should not be included:
- MemoryRegionTestsDSA
- MemoryRegionTestsNoRegion
- UnimplementedTests
- ExtraSpecTests
- IndirectCallTests

PointsToTest probably needs to be removed completely, it's very out of date and doesn't work.

IRTest has been bugged for months (#244) and this isn't running it - the tests that are in a particular package need to have the package prefixed like `./mill test.testOnly ir.IRTest`

BitVectorAnalysisTests and IntrusiveListPublicInterfaceTest are already in the CompileAndTest job so it's redundant to run them multiple times like this.

Reply:
Thank you for the advice, I'll make the changes when I come back :)